File: Java/src/model/PartijaDvaPara.java
package model;

import java.util.Collections;
import java.util.List;

public class PartijaDvaPara extends Partija {

    private List<Igrac> prvi;
    private List<Igrac> drugi;

    public PartijaDvaPara(){
        setVrsta("dvaPara");
    }

    public List<Igrac> getPrvi() {
        return prvi;
    }

    public void setPrvi(List<Igrac> prvi) {
        this.prvi = prvi;
    }

    public List<Igrac> getDrugi() {
        return drugi;
    }

    public void setDrugi(List<Igrac> drugi) {
        this.drugi = drugi;
    }

    @Override
    public boolean isIgraGotova() {
        int ukupnoPrvi=getUkupno(Mjesanje.UKUPNO_PRVI_UNOS),ukupnoDrugi=getUkupno(Mjesanje.UKUPNO_DRUGI_UNOS);
        return ukupnoPrvi==ukupnoDrugi ? false : ukupnoPrvi>getDoKolikoSeIgra() || ukupnoDrugi>getDoKolikoSeIgra();
    }

    @Override
    public String toString() {
        return "Partija DVA PARA, igra gotova: " + isIgraGotova() + ", " + getPrvi().get(0) + " i " + getPrvi().get(1) + ": " + getUkupno(Mjesanje.UKUPNO_PRVI_UNOS) +
                " | "+ getDrugi().get(0) + " i " + getDrugi().get(1)  + ": " + getUkupno(Mjesanje.UKUPNO_DRUGI_UNOS);
    }



}

File: Python/model/PartijaDvaPara.py
from model.Partija import Partija
from model.Rezultat import Rezultat


class PartijaDvaPara(Partija):
    def __init__(self):
        super().__init__()

    def get_rezultat(self) -> Rezultat:
        rezultat = Rezultat()
        rezultat.prvi = sum(mjesanje.get_rezultat().prvi for mjesanje in self.mjesanja)
        rezultat.drugi = sum(mjesanje.get_rezultat().drugi for mjesanje in self.mjesanja)
        return rezultat

    def __str__(self) -> str:
        return 'Partija DVA para' + str(self.is_igra_gotova) + ': ' + str(self.get_rezultat().prvi) + \
               ' - ' + str(self.get_rezultat().drugi)
